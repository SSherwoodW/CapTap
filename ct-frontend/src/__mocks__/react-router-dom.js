import { vi } from "vitest";

export * from "react-router-dom";

export const useParams = vi.fn();
// .. add more mocks if needed
