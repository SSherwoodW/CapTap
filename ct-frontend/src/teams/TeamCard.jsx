import React from "react";
import { Link } from "react-router-dom";



/** Show limited information about a team
 *
 * Is rendered by TeamsList to show a "card" for each team.
 *
 * TeamsList -> TeamCard
 */

function TeamCard({ name, code }) {
  // console.debug("TeamCard", name);

  return (
      <Link className="CompanyCard card" to={`/teams/${code}`}>
        <div className="card-body">
          <h6 className="card-title">
            {code}
          </h6>
          <p>{name}</p>
        </div>
      </Link>
  );
}

export default TeamCard;