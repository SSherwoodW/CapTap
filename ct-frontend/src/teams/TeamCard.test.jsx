import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import TeamCard from "./TeamCard";

describe("TeamCard component", () => {
  it("should render without crashing", () => {
    render(
      <MemoryRouter>
        <TeamCard name="Test Name" code="test" />
      </MemoryRouter>
    );
  });

  it("should match snapshot", () => {
    const { container } = render(
      <MemoryRouter>
        <TeamCard name="Test Name" code="test" />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});