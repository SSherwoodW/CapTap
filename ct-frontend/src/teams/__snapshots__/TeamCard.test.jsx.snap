// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TeamCard component > should match snapshot 1`] = `
<div>
  <a
    class="CompanyCard card"
    href="/teams/test"
  >
    <div
      class="card-body"
    >
      <h6
        class="card-title"
      >
        test
      </h6>
      <p>
        Test Name
      </p>
    </div>
  </a>
</div>
`;
